cmd/minikube/cmd/start.go

```diff
@@ -312,3 +350,73 @@ func init() {
 	viper.BindPFlags(startCmd.Flags())
 	RootCmd.AddCommand(startCmd)
 }
+
+// saveConfig saves profile cluster configuration in
+// $MINIKUBE_HOME/profiles/<profilename>/config.json
+func saveConfig(clusterConfig cluster.Config) error {
+	data, err := json.MarshalIndent(clusterConfig, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	profilePath := filepath.Join(constants.GetMinipath(), "profiles", viper.GetString(cfg.MachineProfile))
+
+	if err := os.MkdirAll(profilePath, 0700); err != nil {
+		return err
+	}
+
+	if err := saveConfigToFile(data, filepath.Join(profilePath, "config.json")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func saveConfigToFile(data []byte, file string) error {
+	if _, err := os.Stat(file); os.IsNotExist(err) {
+		return ioutil.WriteFile(file, data, 0600)
+	}
+
+	tmpfi, err := ioutil.TempFile(filepath.Dir(file), "config.json.tmp")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmpfi.Name())
+
+	if err = ioutil.WriteFile(tmpfi.Name(), data, 0600); err != nil {
+		return err
+	}
+
+	if err = tmpfi.Close(); err != nil {
+		return err
+	}
+
+	if err = os.Remove(file); err != nil {
+		return err
+	}
+
+	if err = os.Rename(tmpfi.Name(), file); err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadConfigFromFile(profile string) (cluster.Config, error) {
+	var cc cluster.Config
+
+	configPath := filepath.Join(constants.GetMinipath(), "profiles", profile, "config.json")
```

Comment:
Can we add this as a "constant" in https://github.com/kubernetes/minikube/blob/master/pkg/minikube/constants/constants.go#L109?  To make it a little easier to find.